cmd/pr.go

```diff
@@ -172,6 +185,7 @@ func (c *create) SanitizeArgs(cCtx *cli.Context) (*args, error) {
 		CheckoutPr:  shouldCheckout,
 		Interactive: cCtx.Bool("interactive"),
 		DraftPr:     cCtx.Bool("draft"),
+		Reset:       cCtx.Bool("reset"),
```

Comment:
I think `--reset` is incompatible with:
- `--interactive` (because the user might not have chosen to pull all the commits in their PR)
- forHead = false (because if you did `opp pr HEAD^` you don't want to lose the last commit)
- localChanges = true (because if there are local changes you don't want to `git reset --hard` them

In addition, we shouldn't reset if the head was not the tip of the main branch (if you checkout pr/2 and add one commit and do `opp pr` there you don't want to have pr/2 reset to master)